ui/src/views/user/survey-edit/SurveyEditView.vue

```diff
@@ -240,22 +249,60 @@ export default {
       return this.formData;
     },
     onSaveAsDraftOptionClick() {
+      console.log("save as draft");
       const finalOutput = this.getData();
       console.log(JSON.stringify(finalOutput));
 
-      this.$notify.toast("Survey has been successfully saved");
+      if(finalOutput.config.startDate == "" || finalOutput.config.endDate == "") {
+        this.$notify.toast("Please give survey's start date and end date!");
+      } else {
+        if (this.currentSurveyId == "new") {
+          userAddSurvey(finalOutput)
+          .then(() => {
+            this.$notify.toast("Your survey has been saved!");
+            this.$router.push("/user/surveys/");
+          }).catch((error) => {
+            this.$notify.toast(error["message"]);
+          });
+        } else {
+          finalOutput.config.id = this.currentSurveyId;
+          userEditSurvey(finalOutput)
+          .then(() => {
+            this.$notify.toast("Your survey has been saved!");
+            window.location.reload();
```

Comment:
Again, why the reload? is it because fetching the data gives some bugs?

Reply:
this morning i just realized that the reloading is not really necessary; we can just call the BE api to save data, pray that data is synched properly between FE and BE, and continue living. I will remove the reload